### PR TITLE
AVP Service

### PR DESCRIPTION
**Pull request by tarrencev**
Initial commit for an AVP service. This is a WIP but would like feedback on the general approach. This implements simple stream recording logic. Eventually i would like to generalize it to support arbitrary processing pipelines. Please let me know if you have any suggestions!

Right now, the AVP process listens to ISLB for new streams, then subscribes to them from the appropriate SFU. In the current implementation, the service only consumes. I would like to extend it to potentially replace the streams in the SFU too, so processed streams can be published to clients.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,29 +1,30 @@
 version: '3.7'
 
 services:
-  web:
```
Comment (tarrencev) [inline docker-compose.yml:4]: I will revert all this

**Comment by Sean-Der** [comment]
:fire: this is amazing! Really nice work @tarrencev 

After work I will get this running locally and play with it. Would you mind if I add you to the Pion organization? Then you can do a branch directly on the repo, will make collab a lot easier.

File: docker/avp.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+WORKDIR $GOPATH/src/github.com/pion/ion/cmd/avp
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /avp .
+
+# FROM alpine:3.9.5
+
+# RUN apk --no-cache add ca-certificates
+# COPY --from=0 /avp /usr/local/bin/avp
+
+# ENTRYPOINT ["/bin/bash"]
+RUN echo 'ping localhost > /dev/null &' > /bootstrap.sh
```
Comment (tarrencev) [inline docker/avp.Dockerfile:23]: this is just to keep the container alive so i can attach and run `avp` during development. Is there a better dev flow?

**Comment by tarrencev** [comment]
Sure that would be great! Awesome. Let me know how it goes. You might need to update the docker depending on your workflow. I kinda hacked things for development 😅

**Comment by tarrencev** [comment]
Would you live me to move this to a branch?

**Comment by adwpc** [comment]
Amazing work! Branch is ok!

**Comment by Sean-Der** [comment]
Sorry I ran out of time to review tonight, promise I will get this done tomorrow :)

I don't know what long term this looks like, but maybe we can have an bucket of 'optional' workers? Then people can easily contribute more stuff like this, and make it as easy as possible to spin them up themselves.

**Comment by tarrencev** [comment]
Closing for now since i have a revised version on a branch ill open for review soon

**Pull request closed without merging**